File: Median of two sorted arrays/Median_of_two_sorted_arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        i, j = 0,0
        res = []
        while i < len(nums1) and j < len(nums2):
            if nums1[i]<=nums2[j]:
                res.append(nums1[i])
                i+=1
            elif nums1[i]>=nums2[j] :
                res.append(nums2[j])
                j+=1
        if i == len(nums1) and j<len(nums2):
            res.extend(nums2[j:len(nums2)])
        if j == len(nums2) and i<len(nums1):
            res.extend(nums1[i:len(nums1)])     


        if len(res)%2 != 0 :
            return res[len(res)//2]
        else : 
            return (res[(len(res)//2)-1]+res[(len(res)//2)])/2
        

File: Median of two sorted arrays/Median_of_two_sorted_arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int i = 0, j = 0;
        List<Integer> res = new ArrayList<>();

        while (i < nums1.length && j < nums2.length) {
            if (nums1[i] <= nums2[j]) {
                res.add(nums1[i]);
                i++;
            } else {
                res.add(nums2[j]);
                j++;
            }
        }
        while (i < nums1.length) {
            res.add(nums1[i]);
            i++;
        }
        while (j < nums2.length) {
            res.add(nums2[j]);
            j++;
        }

        int len = res.size();
        if (len % 2 != 0) {
            return res.get(len / 2);
        } else {
            return (res.get((len / 2) - 1) + res.get(len / 2)) / 2.0;
        }
    }
}
